File: backend/internal/model/recipe.go
package model

import "time"

type Recipe struct {
	ID            uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	Title         string    `gorm:"type:varchar(100);not null;index" json:"title"`
	Cover         string    `gorm:"type:varchar(500)" json:"cover"`
	Description   string    `gorm:"type:text" json:"description"`
	CategoryID    uint      `gorm:"index" json:"category_id"`
	CategoryName  string    `gorm:"-" json:"category_name"`
	CookTime      int       `json:"cook_time"`
	Difficulty    string    `gorm:"type:varchar(20)" json:"difficulty"`
	PeopleCount   int       `json:"people_count"`
	Taste         string    `gorm:"type:varchar(50)" json:"taste"`
	HealthTags    JSON      `gorm:"type:json" json:"health_tags"`
	Ingredients   JSON      `gorm:"type:json" json:"ingredients"`
	Seasonings    JSON      `gorm:"type:json" json:"seasonings"`
	Steps         JSON      `gorm:"type:json" json:"steps"`
	Tips          string    `gorm:"type:text" json:"tips"`
	Nutrition     JSON      `gorm:"type:json" json:"nutrition"`
	ViewCount     int       `gorm:"default:0" json:"view_count"`
	FavoriteCount int       `gorm:"default:0" json:"favorite_count"`
	Status        int8      `gorm:"default:1;index" json:"status"`
	IsFavorited   bool      `gorm:"-" json:"is_favorited"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

func (Recipe) TableName() string { return "recipes" }

File: web-user/src/types/recipe.ts
export interface Recipe {
  id: number
  title: string
  cover: string
  description: string
  cook_time: number
  difficulty: '简单' | '中等' | '困难'
  taste: string
  servings: number
  favorite_count: number
  is_favorited: boolean
  feedback?: RecipeFeedbackStatus
  category_id: number
  category_name: string
  health_tags: string[]
  ingredients: Ingredient[]
  seasonings: Seasoning[]
  steps: CookingStep[]
  nutrition: Nutrition
  tips?: string
}

export interface RecipeFeedbackStatus {
  cooked: boolean
  like: boolean
  dislike: boolean
  block: boolean
  normal: boolean
  too_complex: boolean
  too_long: boolean
  hard_to_buy: boolean
}

export interface Ingredient {
  name: string
  amount: string
  unit: string
  emoji?: string
}

export interface Seasoning {
  name: string
  amount: string
}

export interface CookingStep {
  step: number
  description: string
  image?: string
  tip?: string
}

export interface Nutrition {
  calories: number
  protein: number
  fat: number
  carbs: number
  fiber: number
}

export interface Category {
  id: number
  name: string
  icon: string
  count?: number
}

export interface RecipeQuery {
  keyword?: string
  category_id?: number
  taste?: string
  sort?: 'latest' | 'hot'
  cook_time?: string
  difficulty?: string
  health_tags?: string
  page?: number
  page_size?: number
}
